File: src/main/java/ru/ezhov/rocket/action/api/RocketActionUi.java
package ru.ezhov.rocket.action.api;

import java.awt.Component;

/**
 * UI action builder
 */
public interface RocketActionUi {
    /**
     * Component creation should only happen when this method is called.
     *
     * @return component to display
     */
    Component create(RocketActionSettings settings);

    /**
     * @return action type
     */
    String type();
}

File: src/main/kotlin/ru/ezhov/rocket/action/api/RocketActionUi.kt
package ru.ezhov.rocket.action.api

/**
 * UI action builder
 */
interface RocketActionUi {
    /**
     * Component creation should only happen when this method is called.
     *
     * @return component to display
     */
    fun create(settings: RocketActionSettings): Action

    /**
     * @return action type
     */
    fun type(): String
}